root/etc/cont-init.d/30-config

```diff
@@ -2,38 +2,95 @@
+    BASE_URL=$(echo "${SITE_ROOT}" | sed -E 's/https?:\/\///' | awk -F/ '{ print $1 }' )
 fi
-declare -A HC_CONF
-HC_CONF[DEBUG]=${DEBUG:-False}
-HC_CONF[SITE_ROOT]=${SITE_ROOT:-https://healthchecks.example.com}
-HC_CONF[BASE_URL]=${BASE_URL:-healthchecks.example.com}
-HC_CONF[ALLOWED_HOSTS]=${ALLOWED_HOSTS:-\*}
-HC_CONF[SITE_NAME]=${SITE_NAME:-healthchecks}
-HC_CONF[DEFAULT_FROM_EMAIL]=${DEFAULT_FROM_EMAIL:-noreply@example.com}
-HC_CONF[EMAIL_HOST]=${EMAIL_HOST:-smtp.gmail.com}
-HC_CONF[EMAIL_PORT]=${EMAIL_PORT:-587}
-HC_CONF[EMAIL_HOST_USER]=${EMAIL_HOST_USER-username}
-HC_CONF[EMAIL_HOST_PASSWORD]=${EMAIL_HOST_PASSWORD-password}
-HC_CONF[EMAIL_USE_TLS]=${EMAIL_USE_TLS:-True}
-HC_CONF[REGENERATE_SETTINGS]=${REGENERATE_SETTINGS:-False}
-HC_CONF[SITE_LOGO_URL]=${SITE_LOGO_URL:-}
 
-if [ ! -f "/config/local_settings.py" ] || [[ ${HC_CONF[REGENERATE_SETTINGS]} == True ]]; then
-    # sed in values or skip if value not set
-    for KEY in "${!HC_CONF[@]}"; do
-        if [[ ! ${HC_CONF[$KEY]} == "" ]]; then
-            if [[ ${HC_CONF[$KEY]} == False ]] || [[ ${HC_CONF[$KEY]} == True ]]; then
-                echo "$KEY = ${HC_CONF[$KEY]}" >> /config/local_settings.py
-            else
-                echo "$KEY = \"${HC_CONF[$KEY]}\"" >> /config/local_settings.py
-            fi
+#From https://healthchecks.io/docs/self_hosted_configuration/
+HC_CONF=(
+ALLOWED_HOSTS
+APPRISE_ENABLED
+DEBUG
+DEFAULT_FROM_EMAIL
+DISCORD_CLIENT_ID
+DISCORD_CLIENT_SECRET
+EMAIL_HOST
+EMAIL_HOST_PASSWORD
+EMAIL_HOST_USER
+EMAIL_PORT
+EMAIL_USE_TLS
+EMAIL_USE_VERIFICATION
+LINENOTIFY_CLIENT_ID
+LINENOTIFY_CLIENT_SECRET
+MASTER_BADGE_LABEL
+MATRIX_ACCESS_TOKEN
+MATRIX_HOMESERVER
+MATRIX_USER_ID
+MATTERMOST_ENABLED
+MSTEAMS_ENABLED
+OPSGENIE_ENABLED
+PAGERTREE_ENABLED
+PD_APP_ID
+PD_ENABLED
+PING_BODY_LIMIT
+PING_EMAIL_DOMAIN
+PING_ENDPOINT
+PROMETHEUS_ENABLED
+PUSHBULLET_CLIENT_ID
+PUSHBULLET_CLIENT_SECRET
+PUSHOVER_API_TOKEN
+PUSHOVER_EMERGENCY_EXPIRATION
+PUSHOVER_EMERGENCY_RETRY_DELAY
+PUSHOVER_SUBSCRIPTION_URL
+REGISTRATION_OPEN
+REMOTE_USER_HEADER
+RP_ID
+SHELL_ENABLED
+SIGNAL_CLI_ENABLED
+SITE_LOGO_URL
+SITE_NAME
+SITE_ROOT
+SLACK_CLIENT_ID
+SLACK_CLIENT_SECRET
+SLACK_ENABLED
+SPIKE_ENABLED
+TELEGRAM_BOT_NAME
+TELEGRAM_TOKEN
+TRELLO_APP_KEY
+TWILIO_ACCOUNT
+TWILIO_AUTH
+TWILIO_FROM
+TWILIO_USE_WHATSAPP
+VICTOROPS_ENABLED
+WEBHOOKS_ENABLED
+ZULIP_ENABLED
+)
+
+function insert_config() {
+    if grep -E "^$1 = " /config/local_settings.py &> /dev/null; then
+        sed -i -E "s|^$1 = .*\$|$1 = $2|" /config/local_settings.py
+    else
+        echo "$1 = $2" >> /config/local_settings.py
+    fi
+}
+
+if [ ! -f "/config/local_settings.py" ] || [[ "${REGENERATE_SETTINGS}" == "True" ]]; then
+    touch /config/local_settings.py
+    for CONF in "${HC_CONF[@]}"; do
+        if [[ -n "${!CONF}" ]]; then
+             if [[ "${!CONF}" == "True" ]] || [[ "${!CONF}" == "False" ]] || [[ "${!CONF:0:1}" == "[" ]]; then #booleans or arrays
```

Comment:
One side effect of this change is that array variables like `ALLOWED_HOSTS` **must** now be specified with `[]`. Before, line 33 actually required it to be specified as a string, and it only really supported a single entry, contrary to the documentation